src/test/java/com/example/bank/registration/ClientServiceTest.java

```diff
@@ -0,0 +1,153 @@
+        Assertions.assertThrows(ClientNotFoundException.class, () ->
+                clientService.updateClient(clientRequest));
+    }
+
+    @DisplayName("A parameterized test of updateClient() method")
+    @ParameterizedTest
+    @MethodSource("clientBelow18YearsOldSource")
+    void shouldThrowIllegalArgumentExceptionDuringClientUpdateWhenTheClientIsBelow18YearsOld(
+            Client client,
+            ClientRequest clientRequest) {
+        Long id = client.getId();
+        when(clientRepository.findById(id)).thenReturn(Optional.of(client));
+        Assertions.assertThrows(IllegalArgumentException.class, () ->
+                clientService.updateClient(clientRequest));
+    }
+
+    @DisplayName("A parameterized test of deleteClient() method")
+    @ParameterizedTest
+    @MethodSource("clientSource")
+    void shouldDeleteClientWhenGivenId(Client client) {
+        Long id = client.getId();
+        when(clientRepository.findById(id)).thenReturn(Optional.of(client));
+        doNothing().when(clientRepository).delete(client);
+        clientService.deleteClient(id);
+        verify(clientRepository, times(1)).delete(client);
+    }
+
+    @DisplayName("A parameterized test of deleteClient() method")
+    @ParameterizedTest
+    @MethodSource("clientSource")
+    void shouldThrowClientNotFoundExceptionWhenThereIsNoClientWithId(Client client) {
+        Long id = client.getId();
+        when(clientRepository.findById(id)).thenReturn(Optional.empty());
+        Assertions.assertThrows(ClientNotFoundException.class, () ->
+                clientService.deleteClient(id));
+    }
+
+    static Stream<Arguments> clientBelow18YearsOldSource() {
+        return Stream.of(Arguments.of(clientBelow18YearsOldBuilder(), clientRequestBelow18RequestBuilder()));
+    }
+
+    static Stream<ClientRequest> clientRequestBelow18YearsOldSource() {
+        return Stream.of(clientRequestBelow18RequestBuilder()
+        );
+    }
+
+    static Stream<Client> clientSource() {
+        return Stream.of(clientBuilder(), clientWithId3Builder(), clientAdultWithId5Builder(),
+                client18YearsOldBuilder(), clientBelow18YearsOldBuilder());
+    }
+
+    static Stream<Arguments> clientOver18YearsOldUpdateSource() {
+        return Stream.of(Arguments.of(clientAdultWithId5Builder(), updateClientRequestAdultBuilder(),
+                        resultUpdateClientWithId5Builder()),
+                Arguments.of(client18YearsOldBuilder(), updateClientRequest18YOBuilder(),
+                        resultUpdateClient18YearsOldBuilder()));
+    }
+
+    static Stream<Arguments> clientOver18YearsOldSource() {
+        return Stream.of(Arguments.of(clientAdultWithId5Builder(), clientRequest18YOForRegisterBuilder(),
+                resultOfRegisterClientAdultWithIdBuilder()));
+    }
+
+    static Stream<Arguments> clientRequestOver18YearsOldSource() {
+        return Stream.of(Arguments.of(clientRequest18YOForRegisterBuilder(),
```

Comment:
tych metod z argumentowi jest tak duzo ze w sumie polecam wydzielic to do oddzielnej klasy

Reply:
Wydzieliłem do odrębnej klasy.